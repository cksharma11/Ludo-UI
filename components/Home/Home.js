import React from 'react';
import PropType from 'prop-types';
import Header from '../Header/Header';
import HomeStyles from './Home.style';
import labels from '../../config/labels/labels';
import globalStyles from '../GlobalStyle';

const Home = ({ title, createGameLink, joinGameLink, loadGameLink }) => {
  return (
    <div className="main-container">
      <Header title={title} />
      <div className="options">
        <a href={createGameLink}>{labels.CREATE_GAME}</a>
        <a href={joinGameLink}>{labels.JOIN_GAME}</a>
        <a href={loadGameLink}>{labels.LOAD_GAME}</a>
      </div>
      <style jsx>{HomeStyles}</style>
      <style jsx>{globalStyles}</style>
    </div>
  );
};

Home.propTypes = {
  title: PropType.string.isRequired,
  createGameLink: PropType.string.isRequired,
  joinGameLink: PropType.string.isRequired,
  loadGameLink: PropType.string.isRequired
};

export default Home;
