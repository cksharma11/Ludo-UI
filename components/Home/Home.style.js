import css from 'styled-jsx/css';
import colors from '../../config/styles/colors';

const HomeStyles = css`
  html,
  body {
    width: 100%;
    height: 100%;
  }

  .main-container {
    font-family: fantasy;
    display: flex;
    flex-direction: column;
  }

  .options {
    font-family: fantasy;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    background: ${colors.gray.shade2};
    padding: 100px;
  }

  a {
    border-radius: 2px;
    font-family: fantasy;
    width: 200px;
    height: 30px;
    border: 1px solid black;
    color: ${colors.gray.shade1};
    text-align: center;
    line-height: 2;
    text-decoration: none;
    padding: 5px;
    margin: 5px;
  }
`;

export default HomeStyles;
