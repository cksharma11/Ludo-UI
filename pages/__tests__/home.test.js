import React from 'react';
import { shallow } from 'enzyme';
import Home from '../home';

describe('home', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Home />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
