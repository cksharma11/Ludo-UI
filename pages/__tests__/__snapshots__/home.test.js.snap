// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home should render properly 1`] = `"<Home title=\\"Ludo\\" createGameLink=\\"/createGame\\" joinGameLink=\\"joinGame\\" />"`;
